src/main/java/sky/command/EventCommand.java

```diff
@@ -0,0 +1,108 @@
+
+import sky.TaskList;
+import sky.exception.TextNoMeaningException;
+import sky.storage.History;
+import sky.storage.Storage;
+import sky.task.Event;
+import sky.task.Task;
+
+/**
+ * The EventCommand class deals with adding an event task into taskList.
+ */
+public class EventCommand extends Command {
+    private String fullCommand;
+
+    public EventCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    @Override
+    public String execute(TaskList taskList, Storage storage, History history)
+            throws TextNoMeaningException, IOException {
+        try {
+            String[] arrOfStrings = generateTaskDescriptionAndUserInputDuration();
+            assert arrOfStrings.length > 0 : "arrOfStrings should not be empty.";
+            String taskDescription = arrOfStrings[0];
+            String taskDurationUserInput = arrOfStrings[1];
+            String taskDuration = produceDateAndTimeForEvent(taskDurationUserInput);
+            Task task = new Event(taskDescription, taskDuration);
+            taskList.addTask(task);
+            // Add task into data file
+            storage.append(task.toString());
+            history.addHistoryInTime(taskList);
+            return generateResponse(task, taskList);
+        } catch (IndexOutOfBoundsException e) {
+            throw new TextNoMeaningException("You have either not entered any text after typing event, "
+                    + "or you have positioned your slash wrongly.");
+        } catch (PatternSyntaxException e) {
+            throw new TextNoMeaningException("There is a problem with the regex expression written by the dev.");
+        }
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    private String produceDateAndTimeForEvent(String s) throws TextNoMeaningException {
+        try {
+            String[] arrOfStrings = s.split(" ");
+            if (arrOfStrings.length != 2) {
+                throw new TextNoMeaningException("Provide the date and time after \"/at\""
+                        + " as: \"yyyy/mm/dd XXXX-XXXX\", where XXXX is time in 24-hours.");
+            }
+            String dateGiven = arrOfStrings[0].replaceAll("/", "-");
+            LocalDate d1 = LocalDate.parse(dateGiven);
+            String dateString = d1.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+
+            String timeGiven1 = arrOfStrings[1].substring(0, 2) + ":" + arrOfStrings[1].substring(2, 4);
+            String timeGiven2 = arrOfStrings[1].substring(5, 7) + ":" + arrOfStrings[1].substring(7);
+            LocalTime t1 = LocalTime.parse(timeGiven1);
+            LocalTime t2 = LocalTime.parse(timeGiven2);
+            String timeString1 = t1.format(DateTimeFormatter.ofPattern("h:mma"));
+            String timeString2 = t2.format(DateTimeFormatter.ofPattern("h:mma"));
+            String finalTimeString = timeString1 + "-" + timeString2;
+
+            return dateString + ", " + finalTimeString;
```

Comment:
Is there a way to shorten this code?

Reply:
You are right, there are plenty of ways and reasons to do so. Thanks for pointing it out! Will work on it.